module/data/actor/npc.mjs

```diff
@@ -248,6 +248,19 @@ export default class NPCData extends CreatureTemplate {
 
   /* -------------------------------------------- */
 
+  /**
+   * Prepare movement & senses values derived from race item.
+   */
+  prepareEmbeddedData() {
+    const race = this.parent.items.find(i => i.type === "race");
+    if ( race ) {
+      AttributesFields.prepareRace.call(this, race, { force: true });
+      this.details.type = race.system.type;
```

Comment:
Seems like this could also be done in `prepareRace`, unless I am missing something?

Reply:
Mainly a pedantic reason because `prepareRace` is a method on `AttributesFields`, but the type is within `details` not `attributes` like the rest of the changed values.